### PR TITLE
fix: ensure GC safety for arena-allocated values

### PR DESCRIPTION
**Pull request by jensneuse**
Fixes use-after-free bugs where arena-allocated `Value` structs held references to heap strings that the GC could collect. All string data is now copied onto the arena via a new `arenaString()` helper, object keys are eagerly unescaped at parse time so results live on the arena, and array backing storage uses `arena.SliceAppend`. Adds 18 comprehensive GC safety tests (`arena_gc_test.go`) covering all API surfaces under aggressive GC pressure with the race detector.

## Summary by CodeRabbit

* **New Features**
  * Arena-mode parsing and arena-aware value constructors plus a merge-with-path helper.

* **Tests**
  * Large GC-stress test suite validating parsing, mutations, merges, marshal/round-trips and many edge cases under aggressive GC.

* **Refactor**
  * Core ops rewritten for arena-backed allocation; keys pre-unescaped and lookup simplified; numeric/path helpers streamlined.

* **Breaking Changes**
  * Several public APIs now accept a leading arena/parent parameter; merge null-handling and numeric getter semantics changed.

* **Chores**
  * CI/toolchain bumped; README and docs expanded with arena guidance.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        go: [ '1.25' ]
+        go: [ '1.26' ]
```
Comment (ysmolski) [inline .github/workflows/ci.yml]: is 1.26 required for this PR to work?

File: util.go [hunk 2]
```diff
@@ -62,17 +97,84 @@ func ValueIsNonNull(v *Value) bool {
...
+	case TypeObject:
+		cp.o = deepCopyObject(a, &v.o)
+	case TypeArray:
+		if len(v.a) > 0 {
+			cp.a = arena.AllocateSlice[*Value](a, len(v.a), len(v.a))
+			for i, item := range v.a {
+				cp.a[i] = DeepCopy(a, item)
+			}
+		}
+	// TypeTrue, TypeFalse, TypeNull: only t is needed, already set above.
```
Comment (ysmolski) [inline util.go]: does it mean that we compied trye value somewhere above? I do not see it. Could not find TrueType used in DeepCopy tests too.

File: README.md [hunk 3]
```diff
@@ -1,103 +1,380 @@
...
+result := v.MarshalTo(nil)
+
+runtime.KeepAlive(a) // ensure arena outlives all uses of v
+```
+
+### Don't mix arenas
+
+All values in a single tree should come from the same arena. If `MergeValues`
+returns a value from `b` and `a`/`b` were allocated on different arenas,
+resetting one arena while the other is still in use causes silent corruption.
```
Comment (ysmolski) [inline README.md:334]: Is the check for mixing arenas possible without causing performance degradations?

**Review by ysmolski — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (jensneuse) [reply .github/workflows/ci.yml]: Good catch — no, Go 1.26 is not required. Reverted go.mod, CI matrix, and linter version back to the master defaults (Go 1.25, golangci-lint v2.5.0).

File: util.go (hunk 2 above)
Comment (jensneuse) [reply util.go]: Good question — the \"copy\" happens on line 119: `cp.t = v.t`. For `TypeTrue`, `TypeFalse`, and `TypeNull`, the type tag *is* the entire value (no string data, no child objects/arrays). So copying just `t` is sufficient. I have clarified the comment to make this more obvious.

File: README.md (hunk 3 above)
Comment (jensneuse) [reply README.md:334]: It would be possible in theory (e.g. tag each Value with its arena and assert at Set/Append time), but it would add a pointer-sized field to every Value and a branch on every mutation — so yes, it would hurt performance. The current approach relies on the convention of using `DeepCopy` at boundaries, which keeps the hot path zero-cost.

Comment (jensneuse) [reply README.md:334]: So we assume that the user copies all values onto the same arena when they use the merge, which is what we're doing.

**Review by ysmolski — APPROVED** [review]
LGTM
